### PR TITLE
Add FhirQueryLoader component to Gluestack-ui render

### PR DESCRIPTION
**Pull request by nickvallee**
#181 

Note: one strange thing I noticed is that the spinner coming in from the themed library is different to what it looks like in the docs, so I am not sure what is going on with that.

Screenshot 2024-01-23 at 8 53 04 AM

### CODE REVIEW

File: packages/gluestack-ui/src/r5/feedback/fhir-query-loader.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+type BoxProps = ComponentProps<typeof Box>;
+type CenterProps = ComponentProps<typeof Center>;
+type SpinnerProps = ComponentProps<typeof Spinner>;
+
+export function GlueStackFhirQueryLoader(
+  props: FhirQueryLoaderRendererProps<GlueStackFhirQueryLoaderProps>,
+): ReactElement | null {
+  if (props.isLoading) {
+    return (
+      <Center style={props.style} {...props.rendererProps?.center}>
```
Comment (julienblin) [inline packages/gluestack-ui/src/r5/feedback/fhir-query-loader.tsx:18]: I wonder if the top-level component should be the same for all conditions (namely a Box)?
I am not sure about the consequences for layout...
We can always adjust later.

**Review by julienblin — APPROVED** [review]
Great - Thank you!
With this in we could adjust the expo template: https://github.com/bonfhir/bonfhir/blob/c684318196e2806bdb2f016b9a7d86157883063c/packages/cli/src/templates/expo.ts#L365.

File: packages/gluestack-ui/src/r5/feedback/fhir-query-loader.tsx (hunk 1 above)
Comment (nickvallee) [reply packages/gluestack-ui/src/r5/feedback/fhir-query-loader.tsx:18]: yeah,I mainly followed the Mantine component, which has a center for the loading, and a <Stack /> for the others. Glustack currently has a VStack and HStack, but I didn't see a Stack (which Native Base had), so I just went with Box for now.

**Comment by nickvallee** [comment]
Created an issue, but I can do adjust the template soon so I'll assign it to my self

**Comment by julienblin** [comment]
> Created an issue, but I can do adjust the template soon so I'll assign it to my self

Awesome - thanks!

**Pull request merged**